### PR TITLE
Add support for skipped and ignored tests in JUnit4 (fixes #112)

### LINKED ISSUE #112: [JUnit4] Ignored tests excluded from the report
Bug report

If you add the JUnit Ignore annotation to a test, it stops showing up in the report, instead of being shown with "Pending" or "Unknown" status. The test just disappears.
To reproduce, just ignore a test like [here](https://github.com/aiannucci/allure-junit-example/commit/b960768070dca0ce7ca0d4dac67952350083884a)

| Allure version | 2.2.1 |
| --- | --- |
| Test framework | junit@4.12 |
| Allure integration | allure-junit4@2.0-BETA15 |
| Generate report using | allure-maven@2.8 |

### PR DESCRIPTION
**Pull request by gladnik**
Implement skipped and ignored tests processing.

#### Checklist
- [x] [Sign Allure CLA][cla]
- [x] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

### CODE REVIEW

File: allure-junit4/src/main/java/io/qameta/allure/junit4/AllureJunit4.java [hunk 1]
```diff
@@ -235,4 +235,30 @@ private MessageDigest getMessageDigest() {
     private String getPackage(final Class<?> testClass) {
         return testClass.getPackage().getName();
     }
+
+    private StatusDetails getIgnoredMessage(final Description description) {
+        final Ignore ignore = description.getAnnotation(Ignore.class);
+        return new StatusDetails()
+                .withMessage(ignore == null || ignore.value().isEmpty()
```
Comment (eroshenkoam) [inline allure-junit4/src/main/java/io/qameta/allure/junit4/AllureJunit4.java]: split this line like: 
1. String message = ...
2. withMessage(message)

Comment (gladnik) [inline allure-junit4/src/main/java/io/qameta/allure/junit4/AllureJunit4.java]: done

File: allure-junit4/src/main/java/io/qameta/allure/junit4/AllureJunit4.java [hunk 2]
```diff
@@ -235,4 +235,29 @@ private MessageDigest getMessageDigest() {
     private String getPackage(final Class<?> testClass) {
         return testClass.getPackage().getName();
     }
+
+    private StatusDetails getIgnoredMessage(final Description description) {
+        final Ignore ignore = description.getAnnotation(Ignore.class);
+        final String message = ignore == null || ignore.value().isEmpty()
```
Comment (eroshenkoam) [inline allure-junit4/src/main/java/io/qameta/allure/junit4/AllureJunit4.java]: a little bit more: `Objects.nonNull(ingore)`

Comment (gladnik) [inline allure-junit4/src/main/java/io/qameta/allure/junit4/AllureJunit4.java]: done

**Comment by baev** [comment]
@gladnik thanks!

**Pull request merged**